### PR TITLE
fix: generate correct registry URL

### LINKED ISSUE #1648: Incorrect npm/pnpm/yarn register info when viewed from a package
**Describe the bug**

Pressing the ( i ) button in the title bar shows the register info for npm, pnpm and yarn. The url here is correct. However if you click on a package to view the package details and then click the ( i ) button in the title bar you will see incorrect urls for the register info.

The incorrect url will contain the package's path, for example `https://your.npm.repo/-/web/detail/your-package` instead of `https://your.npm.repo/`

**To Reproduce**

1. Open Verdaccio Web UI
2. Click on any package
3. Click on the ( i ) button in the title
4. Observe incorrect urls

**Expected behavior**

The repository url should always point to the base url.

**Screenshots**

Screenshot of the correct Register Info viewed from the home screen:

image

Screenshot of the incorrect Register Info viewed from the package screen:

image

### PR DESCRIPTION
**Pull request by DanielRuf**
**Type:**
bug

The following has been addressed in the PR:

*  There is a related issue? Yes
*  Unit or Functional tests are included in the PR? Yes

**Description:**
This PR correctly transforms `/`in the devmode to a full URL like in production and generates the correct registry URL.

**Important**: update `combineBaseUrl` in `node_modules/verdaccio/build/lib/utils.js` to the result of https://github.com/verdaccio/verdaccio/pull/1647

```javascript
function combineBaseUrl(protocol, host, prefix) {
  const result = `${protocol}://${host}`;

  const prefixOnlySlash = prefix === '/';
  if (prefix && !prefixOnlySlash) {
    if (prefix.endsWith('/')) {
      prefix = prefix.slice(0, -1);
    }

    if (prefix.startsWith('/')) {
      return `${result}${prefix}`;
    }

    return prefix;
  }

  return result;
}
```

Resolves https://github.com/verdaccio/verdaccio/issues/1648

### CODE REVIEW

**Comment by DanielRuf** [comment]
Please run `yarn build && yarn test:e2e` when reviewing / testing.

File: src/utils/url.test.ts [hunk 1]
```diff
@@ -17,14 +17,18 @@ describe('utils', () => {
       expect(isEmail('')).toBeFalsy();
     });
 
-    test('getRegistryURL() - should keep slash if location is a sub directory', () => {
+    test('getRegistryURL() - should not change when location change', () => {
+      expect(getRegistryURL()).toBe('http://localhost');
       history.pushState({}, 'page title', '/-/web/detail');
-      expect(getRegistryURL()).toBe('http://localhost/-/web/detail');
```
Comment (juanpicado) [inline src/utils/url.test.ts]: Good one

**Review by juanpicado — APPROVED** [review]
Thanks @DanielRuf now looks much better.

**Comment by DanielRuf** [comment]
@juanpicado keep in mind that we might need the fix from verdaccio itself too (regarding https://github.com/verdaccio/verdaccio/pull/1647)

**Comment by juanpicado** [comment]
@DanielRuf do you mean update here the verdaccio dep right?

**Comment by juanpicado** [comment]
It will be on Friday. I just need to close one PR and we are done :)

**Pull request merged**